diff -r b73ccdf1f861 js/src/frontend/Parser.cpp
--- a/js/src/frontend/Parser.cpp	Thu Aug 29 07:50:02 2024 +0000
+++ b/js/src/frontend/Parser.cpp	Mon Sep 09 12:16:30 2024 +0000
@@ -9553,6 +9553,10 @@
 
     // DebuggerStatement
     case TokenKind::Debugger:
+      return handler_.newRawUndefinedLiteral(pos());
+
+    // "real" DebuggerStatement
+    case TokenKind::XDebugger:
       return debuggerStatement();
 
     // |function| is forbidden by lookahead restriction (unless as child
@@ -9796,6 +9800,10 @@
 
     // DebuggerStatement
     case TokenKind::Debugger:
+      return handler_.newRawUndefinedLiteral(pos());
+
+    // "real" DebuggerStatement
+    case TokenKind::XDebugger:
       return debuggerStatement();
 
     // Declaration[Yield]:
diff -r b73ccdf1f861 js/src/frontend/ReservedWords.h
--- a/js/src/frontend/ReservedWords.h	Thu Aug 29 07:50:02 2024 +0000
+++ b/js/src/frontend/ReservedWords.h	Mon Sep 09 12:16:30 2024 +0000
@@ -21,6 +21,7 @@
   MACRO(const, const_, TokenKind::Const)                                  \
   MACRO(continue, continue_, TokenKind::Continue)                         \
   MACRO(debugger, debugger, TokenKind::Debugger)                          \
+  MACRO(xdebugger, xdebugger, TokenKind::XDebugger)                       \
   MACRO(default, default_, TokenKind::Default)                            \
   MACRO(delete, delete_, TokenKind::Delete)                               \
   MACRO(do, do_, TokenKind::Do)                                           \
diff -r b73ccdf1f861 js/src/frontend/TokenKind.h
--- a/js/src/frontend/TokenKind.h	Thu Aug 29 07:50:02 2024 +0000
+++ b/js/src/frontend/TokenKind.h	Mon Sep 09 12:16:30 2024 +0000
@@ -119,6 +119,7 @@
   MACRO(Finally, "keyword 'finally'")                                  \
   MACRO(Throw, "keyword 'throw'")                                      \
   MACRO(Debugger, "keyword 'debugger'")                                \
+  MACRO(XDebugger, "keyword 'xdebugger'")                              \
   MACRO(Export, "keyword 'export'")                                    \
   MACRO(Import, "keyword 'import'")                                    \
   MACRO(Class, "keyword 'class'")                                      \
diff -r b73ccdf1f861 js/src/vm/CommonPropertyNames.h
--- a/js/src/vm/CommonPropertyNames.h	Thu Aug 29 07:50:02 2024 +0000
+++ b/js/src/vm/CommonPropertyNames.h	Mon Sep 09 12:16:30 2024 +0000
@@ -125,6 +125,7 @@
   MACRO_(daysInWeek, "daysInWeek")                                             \
   MACRO_(daysInYear, "daysInYear")                                             \
   MACRO_(debugger, "debugger")                                                 \
+  MACRO_(xdebugger, "xdebugger")                                               \
   MACRO_(decimal, "decimal")                                                   \
   MACRO_(decodeURI, "decodeURI")                                               \
   MACRO_(decodeURIComponent, "decodeURIComponent")                             \
