diff -r b73ccdf1f861 browser/app/profile/firefox.js
--- a/browser/app/profile/firefox.js	Thu Aug 29 07:50:02 2024 +0000
+++ b/browser/app/profile/firefox.js	Mon Sep 09 12:09:48 2024 +0000
@@ -226,7 +226,7 @@
 #endif
 
 // UI tour experience.
-pref("browser.uitour.enabled", true);
+pref("browser.uitour.enabled", false);
 pref("browser.uitour.loglevel", "Error");
 pref("browser.uitour.requireSecure", true);
 pref("browser.uitour.url", "https://www.mozilla.org/%LOCALE%/firefox/%VERSION%/tour/");
