diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 9b56ba74cff12..6e430ced31de0 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -2156,7 +2156,7 @@ pref("browser.migrate.opera-gx.enabled", false);
 pref("browser.migrate.content-modal.enabled", false);
 
 pref("extensions.pocket.api", "api.getpocket.com");
-pref("extensions.pocket.enabled", true);
+pref("extensions.pocket.enabled", false);
 pref("extensions.pocket.oAuthConsumerKey", "40249-e88c401e1b1f2242d9e441c4");
 pref("extensions.pocket.site", "getpocket.com");
 pref("extensions.pocket.onSaveRecs", true);
