diff --git a/browser/components/urlbar/UrlbarInput.sys.mjs b/browser/components/urlbar/UrlbarInput.sys.mjs
index c2b886df81883..177f0d73b1d4b 100644
--- a/browser/components/urlbar/UrlbarInput.sys.mjs
+++ b/browser/components/urlbar/UrlbarInput.sys.mjs
@@ -2883,6 +2883,7 @@ export class UrlbarInput {
    *  Urlbar state, and whether the selection is empty.
    */
   _maybeSelectAll() {
+    return;
     if (
       !this._preventClickSelectsAll &&
       this._compositionState != lazy.UrlbarUtils.COMPOSITION.COMPOSING &&
