diff --git a/browser/components/urlbar/UrlbarInput.jsm b/browser/components/urlbar/UrlbarInput.jsm
index ee03d959d0222..a9ec314e82414 100644
--- a/browser/components/urlbar/UrlbarInput.jsm
+++ b/browser/components/urlbar/UrlbarInput.jsm
@@ -2877,6 +2877,7 @@ class UrlbarInput {
    *  Urlbar state, and whether the selection is empty.
    */
   _maybeSelectAll() {
+    return;
     if (
       !this._preventClickSelectsAll &&
       this._compositionState != lazy.UrlbarUtils.COMPOSITION.COMPOSING &&
