diff --git a/browser/components/search/content/searchbar.js b/browser/components/search/content/searchbar.js
index 3efd14668bfc..b2062e527250 100644
--- a/browser/components/search/content/searchbar.js
+++ b/browser/components/search/content/searchbar.js
@@ -563,6 +563,7 @@
      * searchbar state, and whether the selection is empty.
      */
     _maybeSelectAll() {
+      return;
       if (
         !this._preventClickSelectsAll &&
         document.activeElement == this._textbox &&
diff --git a/browser/components/urlbar/content/UrlbarInput.mjs b/browser/components/urlbar/content/UrlbarInput.mjs
index 319fb6870ec1..e1a38e5adfbf 100644
--- a/browser/components/urlbar/content/UrlbarInput.mjs
+++ b/browser/components/urlbar/content/UrlbarInput.mjs
@@ -5167,6 +5167,7 @@ ${
    *  Urlbar state, and whether the selection is empty.
    */
   #maybeSelectAll() {
+    return;
     if (
       !this.#preventClickSelectsAll &&
       this.#compositionState != UrlbarShared.COMPOSITION.COMPOSING &&
